File: ARCADE/Intro/Level 2_1: adjacentElementsProduct.cs
    using System;
     
    public class Test
    {
    	public static int adjacentElementsProduct(int[] inputArray){
            int max = -1000;
            for (int i = 0; i < inputArray.Length - 1; i++){
                for (int j = i + 1; j < i + 2; j++){
                    if (inputArray[i] * inputArray[j] > max) max = inputArray[i] * inputArray[j];
                }
            }
            return max;
        }
    	public static void Main()
    	{
    		int [] inputArray = new int []{3, 6, -2, -5, 7, 3};
    		Console.WriteLine(adjacentElementsProduct(inputArray));
    	}
    }

File: ARCADE/Intro/Level 2_1: adjacentElementsProduct.pas
program adjacentElementsProduct;
Type  
  ArrayInteger = Array of integer;
  
function adjacentElementsProduct(inputArray: ArrayInteger): integer;
var i, max: integer;
begin
    max := -1000;
    for i := 0 to length(inputArray) - 2 do 
        if(inputArray[i] * inputArray[i+1] >= max) then max := inputArray[i] * inputArray[i+1];
    adjacentElementsProduct := max;
end;
var A : ArrayInteger;
	i : integer;
begin
	A := ArrayInteger.Create(3, 6, -2, -5, 7, 3);
	writeln(adjacentElementsProduct(A))
end.
